docs/specs/clients/core/identity/identity-api.md

```diff
@@ -0,0 +1,21 @@
+# Identity API
+
+
+```typescript
+abstract class IdentityClient {
+  // Registers a blockchain account with an identity key if not yet registered on this client
+  // onSign(message) promise for signing CAIP-122 message to verify blockchain account ownership
+  // returns the public identity key. Method should throw if any errors comes from onSign promise. 
+  public abstract register(params: {
+    account: string;
+    private?: boolean;
+    domain: string,
+    isLimited?: boolean,
```

Comment:
Do we really want `isLimited` instead of just passing `statement` in the params?

Reply:
I assume the SDK is supposed to generate the statement.

Reply:
Yes obviously, but I mean allow passing any statement instead of limiting statement generation by `isLimited` . I think it'll work better long term

Reply:
instead of boolean we could have an enum of any supported statement types. However, at this moment there are only two

Reply:
I'm not too insistent on this but I believe down the line we'll switch to passing a `string`. If majority agree on passing a boolean, I'm okay with it

Reply:
I think it needs to be Identity Client's responsibility to generate the statement